core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -348,9 +348,9 @@ $(document).ready(function () {
 
             // prepare data for executor summary table
             stageExecutorSummaryInfoKeys = Object.keys(responseBody.executorSummary);
+            var executorDetailsMap = {};
             $.getJSON(createRESTEndPointForExecutorsPage(appId),
               function(executorSummaryResponse, status, jqXHR) {
-                var executorDetailsMap = {};
                 executorSummaryResponse.forEach(function (executorDetail) {
                     executorDetailsMap[executorDetail.id] = executorDetail;
```

Comment:
So one issue I think here is that this is being filled in async from what the task table is being filled in and referencing this field.  You could potentially have issues where this hasn't finished by the time the task table wants to use it.

Reply:
Oh sorry about that. I thought the ajax calls was i serial(which makes no sense at all)..
Then I think we should just use a hash map in the backend.